packages/react/src/components/DatePicker/DatePicker.js

```diff
@@ -581,14 +581,20 @@ export default class DatePicker extends Component {
 
     const childArray = React.Children.toArray(children);
     const childrenWithProps = childArray.map((child, index) => {
-      if (index === 0 && child.type === DatePickerInput) {
+      if (
+        index === 0 &&
+        child.type === React.createElement(DatePickerInput).type
```

Comment:
Would we have any prop type warnings if we do `createElement` without required props (I think `id` in this case)?

Reply:
Good catch - Updated.